pytensor/tensor/extra_ops.py

```diff
@@ -283,8 +283,11 @@ class CumOp(COp):
     def __init__(self, axis: int | None = None, mode="add"):
         if mode not in ("add", "mul"):
             raise ValueError(f'{type(self).__name__}: Unknown mode "{mode}"')
-        self.axis = axis
-        self.mode = mode
+        if isinstance(axis, int) or axis is None:
```

Comment:
Nitpick, it's more common to just check and raise than indenting the "correct code"

```suggestion
        if not isinstance(axis, int) or axis is None:
            ### raise error
        # usual code
```